Documentation/security/policy/standalone-dns-proxy.rst

```diff
@@ -0,0 +1,285 @@
+       Server:         10.96.0.10
+       Address:        10.96.0.10#53
+
+       ** server can't find example.com: REFUSED
+
+    Both queries demonstrate that the standalone DNS proxy continues to enforce DNS policies
+    independently, allowing ``cilium.io`` and refusing ``example.com``, even when the agent is unavailable.
+
+11. **Restore Cilium Agent**
+
+    Restore the Cilium agent to normal operation:
+
+    .. code-block:: shell-session
+
+       $ kubectl rollout undo ds/cilium -n kube-system
+
+For more information on DNS policies, see :ref:`DNS based`.
+
+Limitations
+===========
+
+The Standalone DNS Proxy alpha release has the following known limitations:
+
+* The standalone DNS proxy uses the agent's DNS configuration (``dnsProxy.proxyPort`` 
+  and ``dnsProxy.enableDnsCompression``). The ``dnsProxy.proxyPort`` must be explicitly 
+  set to a non-zero value when the standalone DNS proxy is enabled. Currently, it does not select
+  a free port if not set.
+* Metrics related to DNS are not supported yet. The metrics are currently
+  only available from the in-agent DNS proxy.
+* Standalone DNS proxy depends on cilium agent to read DNS policies, enforce them and 
+  communicate via gRPC. If there are connectivity issues between the proxy and agent,
+  DNS policy enforcement may be affected.
+* While the standalone DNS proxy can continue to proxy DNS requests when the agent is down,
+  it cannot allocate new identities for domains that haven't been observed before.
+  If an endpoint looks up a new domain (one that hasn't been cached) while the agent is unavailable,
+  the resulting traffic will be dropped because no security identity can be allocated.
+  Only DNS lookups for previously observed domains (with cached identities) will work during agent downtime.
+
+Troubleshooting
+===============
+
+Validation Errors
+-----------------
+
+If the Helm chart fails with a validation error about ``dnsProxy.proxyPort``:
+
+.. code-block:: text
+
+   Error: standaloneDnsProxy requires dnsProxy.proxyPort to be set to a non-zero value (e.g., 10094)
+
+This means you need to explicitly configure the DNS proxy port in your values:
+
+.. code-block:: yaml
+
+   dnsProxy:
+     proxyPort: 10094  # Must be non-zero
+
+Port Configuration Verification
+--------------------------------
+
+To verify that the DNS proxy port is correctly configured:
+
+.. code-block:: shell-session
+
+   $ kubectl -n kube-system get configmap cilium-config -o yaml | grep 'tofqdns-proxy-port'
+   $ kubectl -n kube-system get configmap standalone-dns-proxy-config -o yaml | grep 'tofqdns-proxy-port'
+
+Both ConfigMaps should show the same value from ``dnsProxy.proxyPort``.
+
+gRPC Communication Issues
+-------------------------
+
+If there are communication issues between the proxy and agent, review agent logs for connection errors:
+
+.. code-block:: shell-session
+
+   $ kubectl -n kube-system logs -l k8s-app=cilium --tail=100 | grep -i "grpc"
```

Comment:
Grepping for the correct `subsys` might also improve the output here, since other components like Hubble also use gRPC.